src/services/goalSimilarityGroup.ts

```diff
@@ -256,7 +257,7 @@ export async function createSimilarityGroup(
     );
   }
 
-  if (group) {
+  if (!byPassChecks && group) {
```

Comment:
Looks good assuming we are cleaning up any groups that might already have these goals.